From 569ef7bf521d6f184d4f988cc676f127b4da422a Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 8 Oct 2019 14:58:07 +0700
Subject: [PATCH 196/351] gpio: gpio-rcar: gpio-rcar: prevent to force the
 interrupt handler as thread

On RT system, when kernel is booting with CONFIG_IRQ_FORCED_THREADING
flag, the gpio_rcar_irq_handler() will be forced as threaded and then
will generate a WARNING MESSAGE related to interrupt handlers.
To prevent to force the interrupt handler as threaded, we will add
IRQF_NO_THREAD flag when requesting an interrupt for gpio-rcar driver.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 drivers/gpio/gpio-rcar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-rcar.c b/drivers/gpio/gpio-rcar.c
index c8d526a..e372dbb 100644
--- a/drivers/gpio/gpio-rcar.c
+++ b/drivers/gpio/gpio-rcar.c
@@ -512,7 +512,7 @@ static int gpio_rcar_probe(struct platform_device *pdev)
 
 	p->irq_parent = irq->start;
 	if (devm_request_irq(dev, irq->start, gpio_rcar_irq_handler,
-			     IRQF_SHARED, name, p)) {
+			     IRQF_SHARED | IRQF_NO_THREAD, name, p)) {
 		dev_err(dev, "failed to request IRQ\n");
 		ret = -ENOENT;
 		goto err1;
-- 
2.7.4

